pkg/scheduler/metrics/profile_metrics.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"time"
+
+	"k8s.io/component-base/metrics"
+)
+
+var (
+	scheduledResult     = "scheduled"
+	unschedulableResult = "unschedulable"
+	errorResult         = "error"
+)
+
+// PerProfile holds metrics that are associated to a profile and provides
+// helper functions for observing values.
+type PerProfile struct {
+	perResult map[string]perResult
+}
+
+// NewPerProfile creates a new PerProfile that holds profile metrics.
+func NewPerProfile(name string) *PerProfile {
+	perResultM := make(map[string]perResult)
+	for _, result := range []string{scheduledResult, unschedulableResult, errorResult} {
+		labels := map[string]string{
+			"result":  result,
+			"profile": name,
+		}
+		perResultM[result] = perResult{
+			counter:  scheduleAttempts.With(labels),
```

Comment:
So here you could just do:

```
perResultM[result] = perResult{
    counter:  scheduleAttempts.WithLabelValues(result, profile),
    duration: e2eSchedulingLatency.WithLabelValues(result, profile),
}
```

Reply:
but that's mostly a stylistic preference thing. I'm actually pretty ambivalent towards using one or the other (though if a metric has many labels, it tends to make more sense to use the map since it is much easier to mess with the ordering).